app/controllers/transactions_controller.rb

```diff
@@ -0,0 +1,27 @@
+class TransactionsController < ApplicationController
+  def index
+    category = Category.includes(:transactions).find(params[:category_id])
+    @transactions = category.transactions.order(:created_at)
+    @total = category.transactions.sum(:price)
+  end
+
+  def new
+    @transaction = Transaction.new
+  end
+
+  def create
+    @transaction = Transaction.new(transaction_params)
+    @transaction.category_id = params[:category_id]
+    if @transaction.save!
+      redirect_to category_transactions_path, notice: 'Transaction was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def transaction_params
+    params.require(:transaction).permit(:name, :description, :image, :price)
+  end
+end
```

Comment:
- [ ] here the flash mesage does not leave until I refresh hereby blocking from clicking the button, kindly fix
![Screenshot from 2023-01-20 23-19-28](https://user-images.githubusercontent.com/66097371/213818904-442703f2-c14e-427d-87fd-622f0ecde96f.png)